### PR TITLE
[DOCS] Update the home page of documentation to keep the latest updates

### PR DESCRIPTION
**Pull request by yaooqinn**
### _Why are the changes needed?_

The doc home page is stale now for the scope of the Kyuubi community and project as it is written based on spark SQL scenario only.

This PR changes the kyuubi_layers.drawio to add missing Doris, Flink, and Hive logos.

And other changes can be viewed from the screenshot pinned below.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

image

image

image
image

- [ ] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by yaooqinn** [comment]
cc @apache/kyuubi-committers

**Comment by ulysses-you** [comment]
looks good

**Comment by yaooqinn** [comment]
cc @deadwind4 @bowenliang123 too

File: docs/conf.py [hunk 1]
```diff
@@ -117,9 +117,12 @@
     "repository_branch": "master",
     "path_to_docs": "docs",
     "logo_only": True,
-    "home_page_in_toc": True,
+    "home_page_in_toc": False,
```
Comment (yaooqinn) [inline docs/conf.py:120]: this is set to false as we can click the kyuubi logo to go home page

**Review by pan3793 — APPROVED** [review]

**Review by turboFei — APPROVED** [review]

**Review by cxzl25 — APPROVED** [review]

**Comment by turboFei** [comment]
fix the ci issue, could you rebase this pr on master?

File: docs/index.rst [hunk 2]
```diff
@@ -18,79 +18,160 @@
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+End users can use the application programming interface listed below for connectivity
+and interoperation between supported clients and a Kyuubi server. The current implementations are:
+
+- Hive Thrift Protocol
+   - A HiveServer2-compatible interface that allows end users to use a thrift
+     client(cross-language support, both tcp and http), a `Java Database Connectivity(JDBC) <client/jdbc/index.html>`_ interface over
+     thrift, or an `Open Database Connectivity (ODBC) <client/odbc/index.html>`_ interface over a JDBC-to-ODBC bridge to communicate with Kyuubi.
+- MySQL Protocol
```
[the remark this answers is not in the document]
Comment (yaooqinn) [reply docs/index.rst:77]: make sense to me, MySQL is not ready, and kind of a maintenance burden

File: docs/index.rst [hunk 3]
```diff
@@ -18,79 +18,160 @@
...
+
+Apache Kyuubi™ is a distributed and multi-tenant gateway to provide serverless
+SQL on Data Warehouses and Lakehouses.
+
+Kyuubi builds distributed SQL query engines on top of various kinds of modern
+computing frameworks, e.g.,
+Apache `Spark <https://spark.apache.org/>`_,
+`Flink <https://flink.apache.org/>`_,
+`Doris <https://doris.apache.org/>`_,
+`Hive <https://hive.apache.org/>`_,
```
Comment (bowenliang123) [inline docs/index.rst:32]: Is it necessary to add Apache in front of Flink, Doris, and Hive, considering they are Apache's trade mark? Or just remove all the `Apache` in this list including for Spark.

Comment (yaooqinn) [reply docs/index.rst:32]: It's ok to keep it as-is

**Comment by yaooqinn** [comment]
Thanks all for the verification. There is still a lot of work that needs to be done for the docs

File: docs/index.rst (hunk 3 above)
Comment (bowenliang123) [reply docs/index.rst:32]: alright

**Comment by yaooqinn** [comment]
merged to master

**Pull request closed without merging**